tests/CMakeLists.txt

```
@@ -167,10 +170,14 @@ file(GLOB testprogs testprogs/*.c)
 foreach(testprog ${testprogs})
   get_filename_component(bin_name ${testprog} NAME_WE)
   add_executable (${bin_name} ${testprog})
+
+  set_property(TARGET ${bin_name} PROPERTY POSITION_INDEPENDENT_CODE FALSE)
+  target_compile_options(${bin_name} PRIVATE -no-pie)
+
   if(HAVE_SYSTEMTAP_SYS_SDT_H)
     target_compile_definitions(${bin_name} PRIVATE HAVE_SYSTEMTAP_SYS_SDT_H)
   endif(HAVE_SYSTEMTAP_SYS_SDT_H)
-  set_target_properties( ${bin_name} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testprogs/ COMPILE_FLAGS "-g -O0")
+  set_target_properties( ${bin_name} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testprogs/ COMPILE_FLAGS "-g -O0 -no-pie")
```

I don't know much about CMake, but I think -no-pie link flag is needed.

```suggestion
  set_target_properties( ${bin_name} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testprogs/ COMPILE_FLAGS "-g -O0" LINK_FLAGS "-no-pie")
```

(It seems  `POSITION_INDEPENDENT_CODE` affects link flags only in [CMake 3.14 and newer](https://cmake.org/cmake/help/latest/policy/CMP0083.html))